sqlmesh/dbt/model.py

```diff
@@ -256,27 +275,44 @@ def model_kind(self, context: DbtContext) -> ModelKind:
 
         incremental_kind_kwargs["on_destructive_change"] = on_destructive_change
         incremental_kind_kwargs["on_additive_change"] = on_additive_change
-        for field in ("forward_only", "auto_restatement_cron"):
-            field_val = getattr(self, field, None)
-            if field_val is None:
-                field_val = self.meta.get(field, None)
-            if field_val is not None:
-                incremental_kind_kwargs[field] = field_val
+        auto_restatement_cron_value = self._get_field_value("auto_restatement_cron")
+        if auto_restatement_cron_value is not None:
+            incremental_kind_kwargs["auto_restatement_cron"] = auto_restatement_cron_value
 
         if materialization == Materialization.TABLE:
             return FullKind()
         if materialization == Materialization.VIEW:
             return ViewKind()
         if materialization == Materialization.INCREMENTAL:
             incremental_by_kind_kwargs: t.Dict[str, t.Any] = {"dialect": self.dialect(context)}
+            forward_only_value = self._get_field_value("forward_only")
+            if forward_only_value is not None:
+                incremental_kind_kwargs["forward_only"] = forward_only_value
+
+            is_incremental_by_time_range = self.time_column or (
+                self.incremental_strategy and self.incremental_strategy == "microbatch"
+            )
+            # Get shared incremental by kwargs
             for field in ("batch_size", "batch_concurrency", "lookback"):
-                field_val = getattr(self, field, None)
-                if field_val is None:
-                    field_val = self.meta.get(field, None)
+                field_val = self._get_field_value(field)
                 if field_val is not None:
+                    # Check if `batch_size` is representing an interval unit and if so that will be handled at the model level
+                    if field == "batch_size" and isinstance(field_val, str):
+                        continue
                     incremental_by_kind_kwargs[field] = field_val
 
-            if self.time_column:
+            disable_restatement = self.disable_restatement
+            if disable_restatement is None:
+                if is_incremental_by_time_range:
+                    disable_restatement = False
+                else:
+                    disable_restatement = (
+                        not self.full_refresh if self.full_refresh is not None else False
+                    )
+            incremental_by_kind_kwargs["disable_restatement"] = disable_restatement
```

Comment:
`disable_restatement` is part of `IncrementBy` so it should be part of `incremental_by_kind_kwargs` and this PR adds it to it.